### PR TITLE
[ENHANCEMENT] Realworld Weather Sync (Only USA)

### PR DESCRIPTION
**Pull request by BrianxTu**
**Describe Pull request**
This adds support for real-world weather. Using weather.gov's api, and lat/long to perform 2 http requests to get the base information needed.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? **[yes]**
- Does your code fit the style guidelines? **[yes]**
- Does your PR fit the contribution guidelines? **[yes]**

### CODE REVIEW

**Comment by AnishBplayz** [comment]
Will it sync server side or client side ?

**Comment by CptNoHand** [comment]
Time sync would be nice aswell.

File: server/server.lua [hunk 1]
```diff
@@ -40,20 +40,62 @@ end
...
-        elseif new == 2 then CurrentWeather = "CLOUDS"
-        elseif new == 3 then CurrentWeather = "CLEAR"
-        elseif new == 4 then CurrentWeather = "EXTRASUNNY"
-        elseif new == 5 then CurrentWeather = "SMOG"
-        else CurrentWeather = "FOGGY"
+    if Config.UseRealWeather then
+        local setupLink = "https://api.weather.gov/points/"..Config.Latitude..","..Config.Longitude
+        local wWeather
+        PerformHttpRequest(setupLink, function(err, rText, headers)
+            local temp = json.decode(rText)
```
Comment (Z3rio) [inline server/server.lua:47]:
```suggestion
        PerformHttpRequest(setupLink, function(err, rText)
            if err and err ~= 0 then
                print("Error code: " .. err)
            end
            local temp = json.decode(rText)
```
Fix linting issue

File: server/server.lua [hunk 2]
```diff
@@ -40,20 +40,62 @@ end
...
+            PerformHttpRequest(temp["properties"]["forecastHourly"], function(err2, rText2, headers2)
+                local temp2 = json.decode(rText2)
```
Comment (Z3rio) [inline server/server.lua:49]:
```suggestion
            PerformHttpRequest(temp["properties"]["forecastHourly"], function(err2, rText2)
                local temp2 = json.decode(rText2)
                if err2 and err2 ~= 0 then
                    print("Error code: " .. err2)
                end
```
Fix linting

**Review by Z3rio — CHANGES_REQUESTED** [review]
This should fix your linting :)

**Review by BrianxTu — COMMENTED** [review]
The changes made are all good with me. Thanks for your time.

**Review by iktray — APPROVED** [review]

**Pull request closed without merging**